### PR TITLE
if: modernize

### PR DESCRIPTION
**Pull request by bl-ue**
- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

### CODE REVIEW

File: pages/common/if.md [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
+- Check if a directory exists:
 
-`if [[ -z $GIT_BRANCH ]]; then echo "true"; else echo "false"; fi`
+`if [[ -d {{path/to/directory}} ]]; then {{echo "true"}}; else {{echo "false"}}; fi`
 
-- Test if a file exists:
+Check if a file or directory ([e]ntry) exists:
 
-`if [[ -e {{filename}} ]]; then echo "true"; else echo "false"; fi`
+`if [[ -e {{path/to/file_or_directory}} ]]; then {{echo "true"}}; else {{echo "false"}}; fi`
```
Comment (ericbn) [inline pages/common/if.md:24]: I think the`-e` here refers to "exists", more than to "entry"...

File: pages/common/if.md [hunk 2]
```diff
@@ -1,27 +1,28 @@
...
 
-- If directory not exists:
+- List all possible conditions (`test` is an alias to `[`; both are commonly used with `if`):
 
-`if [[ ! -d {{path/to/directory}} ]]; then echo "true"; else echo "false"; fi`
+`help test`
```
Comment (ericbn) [inline pages/common/if.md:28]: `help` is specific to bash. I suggest using `man [` here.

File: pages/common/if.md (hunk 1 above)
Comment (bl-ue) [reply pages/common/if.md:24]: Hmm...you must be right 👍🏻

File: pages/common/if.md [hunk 3]
```diff
@@ -1,27 +1,28 @@
...
+`if [[ -f {{path/to/file}} ]]; then {{echo "true"}}; else {{echo "false"}}; fi`
 
-- Test if a given variable is empty:
+- Check if a directory exists:
 
-`if [[ -z $GIT_BRANCH ]]; then echo "true"; else echo "false"; fi`
+`if [[ -d {{path/to/directory}} ]]; then {{echo "true"}}; else {{echo "false"}}; fi`
 
-- Test if a file exists:
+Check if a file or directory ([e]ntry) exists:
```
Comment (bl-ue) [inline pages/common/if.md:22]:
```suggestion
Check if a file or directory exists:
```

File: pages/common/if.md [hunk 4]
```diff
@@ -1,27 +1,28 @@
 # if
 
 > Simple shell conditional.
+> See also: `test`, `[`.
 
-- Echo a different thing depending on a command's success:
+- Executed two different commands based on a condition:
 
-`if {{command}}; then echo "success"; else echo "failure"; fi`
```
Comment (ericbn) [inline pages/common/if.md:7]: I would keep this, to show that if can be used with any command, not only with `[` or `test`...

File: pages/common/if.md (hunk 2 above)
Comment (bl-ue) [reply pages/common/if.md:28]: Oh yeah, you're right about that!  I was going to do something about that but I was distracted...:smile:
```suggestion
`man [`
```
My shell `zsh` doens't even have help :P

File: pages/common/if.md (hunk 4 above)
Comment (bl-ue) [reply pages/common/if.md:7]: Well, I removed it because it's nearly a duplicate of the next example :/

File: pages/common/if.md [hunk 5]
```diff
@@ -1,27 +1,28 @@
...
+`if {{condition}}; then {{echo "true"}}; else {{echo "false"}}; fi`
 
-- Full if syntax:
+- Check if a variable is defined:
 
-`if {{condition}}; then echo "true"; else echo "false"; fi`
+`if [[ ! -z {{$VARIABLE}} ]]; then {{echo "defined"}}; else {{echo "not defined"}}; fi`
```
Comment (ericbn) [inline pages/common/if.md:12]: `! -z` won't test if a variable is defined, but more specifically if the the length of the variable value is non-zero. This is the same as using `-n`, which I suggest should be used instead.

Comment (bl-ue) [reply pages/common/if.md:12]: I didn't know about `-n`. 👍🏻   Yes, I know that it doesn't actually check if it's defined, but to a user, what's the difference between a variable that's not defined and a variable that has no value?
```suggestion
`if [[ -n {{$VARIABLE}} ]]; then {{echo "defined"}}; else {{echo "not defined"}}; fi`
```

File: pages/common/if.md (hunk 4 above)
Comment (ericbn) [reply pages/common/if.md:7]: Then maybe just keep this one. `if` does not run on a "condition", but rather on any command (built-in or not)...

Comment (bl-ue) [reply pages/common/if.md:7]: Ah good point.  I should have tested that 😄

Comment (bl-ue) [reply pages/common/if.md:7]: Fixed https://github.com/tldr-pages/tldr/pull/5351/commits/c46b7b93d6fcd420e1ce6670bc4d07ebe3865f54

File: pages/common/if.md (hunk 5 above)
Comment (ericbn) [reply pages/common/if.md]: A variable can be defined or not, and have an empty value or not. These are two different things for a shell. This works in zsh:
```
% unset FOO
% if (( ${+FOO} )); then echo "FOO is defined"; else echo "FOO is not defined"; fi
FOO is not defined
% FOO=
% if (( ${+FOO} )); then echo "FOO is defined"; else echo "FOO is not defined"; fi
FOO is defined
% if [[ -n ${FOO} ]]; then echo "FOO is nonempty"; else echo "FOO is empty"; fi
FOO is empty
% FOO=1
% if [[ -n ${FOO} ]]; then echo "FOO is nonempty"; else echo "FOO is empty"; fi
FOO is nonempty
```
I know it's tricky...  :- )

File: pages/common/if.md [hunk 6]
```diff
@@ -1,27 +1,28 @@
...
+- Executed two different commands based on a condition:
 
-`if {{command}}; then echo "success"; else echo "failure"; fi`
+`if {{command}}; then {{echo "true"}}; else {{echo "false"}}; fi`
 
-- Full if syntax:
+- Check if a variable is defined:
 
-`if {{condition}}; then echo "true"; else echo "false"; fi`
+`if [[ -n {{$VARIABLE}} ]]; then {{echo "defined"}}; else {{echo "not defined"}}; fi`
```
Comment (ericbn) [inline pages/common/if.md:12]: A variable can be defined or not, and have an empty value or not. These are two different things for a shell. This works in zsh:
```
% unset FOO
% if (( ${+FOO} )); then echo "FOO is defined"; else echo "FOO is not defined"; fi
FOO is not defined
% FOO=
% if (( ${+FOO} )); then echo "FOO is defined"; else echo "FOO is not defined"; fi
FOO is defined
% if [[ -n ${FOO} ]]; then echo "FOO is nonempty"; else echo "FOO is empty"; fi
FOO is empty
% FOO=1
% if [[ -n ${FOO} ]]; then echo "FOO is nonempty"; else echo "FOO is empty"; fi
FOO is nonempty
```
I know it's tricky...  :- )

**Comment by bl-ue** [comment]
Thanks for the review @ericbn! 🙂

**Review by navarroaxel — APPROVED** [review]

File: pages/common/if.md [hunk 7]
```diff
@@ -1,27 +1,28 @@
...
 
-- List available if conditions:
+- Check if a file exists:
 
-`help test`
+`if [[ -f {{path/to/file}} ]]; then {{echo "true"}}; else {{echo "false"}}; fi`
```
Comment (sbrl) [inline pages/common/if.md:16]: Same here.
```suggestion
`if [[ -f "{{path/to/file}}" ]]; then {{echo "true"}}; else {{echo "false"}}; fi`
```

File: pages/common/if.md [hunk 8]
```diff
@@ -1,27 +1,28 @@
...
 
-- Test if a given variable is empty:
+- Check if a directory exists:
 
-`if [[ -z $GIT_BRANCH ]]; then echo "true"; else echo "false"; fi`
+`if [[ -d {{path/to/directory}} ]]; then {{echo "true"}}; else {{echo "false"}}; fi`
```
Comment (sbrl) [inline pages/common/if.md:20]: Same here.
```suggestion
`if [[ -d "{{path/to/directory}}" ]]; then {{echo "true"}}; else {{echo "false"}}; fi`
```

File: pages/common/if.md [hunk 9]
```diff
@@ -1,27 +1,28 @@
...
 
-- Test if a file exists:
+Check if a file or directory exists:
 
-`if [[ -e {{filename}} ]]; then echo "true"; else echo "false"; fi`
+`if [[ -e {{path/to/file_or_directory}} ]]; then {{echo "true"}}; else {{echo "false"}}; fi`
```
Comment (sbrl) [inline pages/common/if.md:24]: and here.
```suggestion
`if [[ -e "{{path/to/file_or_directory}}" ]]; then {{echo "true"}}; else {{echo "false"}}; fi`
```

**Review by sbrl — COMMENTED** [review]
Thanks for the new page! I've left some comments below for you to review.

**Review by sbrl — APPROVED** [review]
Thanks, @bl-ue!

**Comment by sbrl** [comment]
That's not a very helpful error message, @tldr-bot.

File: pages/common/if.md [hunk 10]
```diff
@@ -1,27 +1,28 @@
...
+`if [[ -f "{{path/to/file}}" ]]; then {{echo "true"}}; else {{echo "false"}}; fi`
 
-- Test if a given variable is empty:
+- Check if a directory exists:
 
-`if [[ -z $GIT_BRANCH ]]; then echo "true"; else echo "false"; fi`
+`if [[ -d "{{path/to/directory}}" ]]; then {{echo "true"}}; else {{echo "false"}}; fi`
 
-- Test if a file exists:
+Check if a file or directory exists:
```
Comment (bl-ue) [inline pages/common/if.md:22]:
```suggestion
- Check if a file or directory exists:
```

**Review by bl-ue — COMMENTED** [review]
@sbrl, indeed no it isn't! But this is why:

**Review by sbrl — APPROVED** [review]

**Pull request merged**